x-pack/plugins/observability_solution/observability_ai_assistant/public/hooks/use_user_preferred_language.ts

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { useCallback } from 'react';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+
+const USE_KIBANA_LOCALE_SETTING = 'Use Kibana locale setting';
+const USE_BROWSER_LANGUAGE_SETTING = 'Use browser language setting';
+
+const SPECIAL_OPTIONS = [
+  {
+    label: USE_KIBANA_LOCALE_SETTING,
```

Comment:
Should there perhaps be a (human readable) label and a value?

```suggestion
    label: i18n('Use Kibana locale setting'),
    value: USE_KIBANA_LOCALE_SETTING
```

Reply:
I tried this while building this but I just couldn't get this to work. 
The docs agree with you and say it should be possible to have label+value but I couldn't get it to work, in that it would chose the wrong option to be selected. 
Eventually I gave up and just used the label, but that also means I only have the label to compare on when checking which option is selected so it cannot be translated.